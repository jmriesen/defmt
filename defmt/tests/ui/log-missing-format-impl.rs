struct Foo;

fn main() {
    defmt::info!("{}", Foo)
}
