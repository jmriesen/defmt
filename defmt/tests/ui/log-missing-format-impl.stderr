error[E0277]: the trait bound `Foo: Format` is not satisfied
 --> tests/ui/log-missing-format-impl.rs:4:5
  |
4 |     defmt::info!("{}", Foo)
  |     ^^^^^^^^^^^^^^^^^^^^^^^ the trait `Format` is not implemented for `Foo`
  |
  = help: the following other types implement trait `Format`:
            &T
            &mut T
            ()
            (T0, T1)
            (T0, T1, T2)
            (T0, T1, T2, T3)
            (T0, T1, T2, T3, T4)
            (T0, T1, T2, T3, T4, T5)
          and $N others
note: required by a bound in `defmt::export::fmt`
 --> src/export/mod.rs
  |
  | pub fn fmt<T: Format + ?Sized>(f: &T) {
  |               ^^^^^^ required by this bound in `fmt`
  = note: this error originates in the macro `defmt::info` (in Nightly builds, run with -Z macro-backtrace for more info)
